apps/explorer-e2e/src/specs/contract.spec.ts

```diff
@@ -6,55 +6,128 @@ import {
 let cluster: Cluster
 
-test.beforeEach(async ({ page, context }) => {
-  cluster = await startCluster({ context })
-  await renterdWaitForContracts({
-    renterdNode: cluster.daemons.renterds[0].node,
-    hostdCount: cluster.daemons.hostds.length,
-  })
+const networkVersions = ['v1', 'v2', 'transition'] as const
 
-  await exploredStabilization(cluster)
+for (const networkVersion of networkVersions) {
+  test.describe(`${networkVersion}`, () => {
+    test.beforeEach(async ({ page, context }) => {
+      cluster = await startCluster({ context, networkVersion })
+      await renterdWaitForContracts({
+        renterdNode: cluster.daemons.renterds[0].node,
+        hostdCount: cluster.daemons.hostds.length,
+      })
+      await exploredStabilization(cluster)
+      explorerApp = new ExplorerApp(page)
+    })
 
-  explorerApp = new ExplorerApp(page)
-})
+    test.afterEach(async () => {
+      await teardownCluster()
+    })
 
-test.afterEach(async () => {
-  await teardownCluster()
-})
+    // Tests
+    test('contract can be searched by id', async ({ page }) => {
+      /* eslint-disable playwright/no-conditional-in-test */
+      if (networkVersion === 'v1') {
+        const hostContracts = await cluster.daemons.hostds[0].api.contracts({
+          data: { statuses: ['active'] },
+        })
+        const contractID = hostContracts.data.contracts[0].revision.parentID
 
-test('contract can be searched by id', async ({ page }) => {
-  const hostContracts = await cluster.daemons.hostds[0].api.contracts({
-    data: { statuses: ['active'] },
-  })
-  const contractID = hostContracts.data.contracts[0].revision.parentID
+        await explorerApp.goTo('/')
+        await explorerApp.navigateBySearchBar(contractID)
 
-  await explorerApp.goTo('/')
-  await explorerApp.navigateBySearchBar(contractID)
+        await expect(
+          page
+            .getByTestId('entity-heading')
+            .getByText('Contract ' + contractID.slice(0, 5))
+        ).toBeVisible()
+      }
 
-  await expect(
-    page
-      .getByTestId('entity-heading')
-      .getByText('Contract ' + contractID.slice(0, 5))
-  ).toBeVisible()
-})
+      if (networkVersion === 'transition') {
+        const hostContracts = await cluster.daemons.hostds[0].api.contractsV2({
+          data: { statuses: ['active'] },
+        })
+        const contractID = hostContracts.data.contracts[0].id
 
-test('contract can be directly navigated to', async ({ page }) => {
-  const hostContracts = await cluster.daemons.hostds[0].api.contracts({
-    data: { statuses: ['active'] },
-  })
-  const contractID = hostContracts.data.contracts[0].revision.parentID
+        await explorerApp.goTo('/')
+        await explorerApp.navigateBySearchBar(contractID)
+
+        await expect(
+          page
+            .getByTestId('entity-heading')
+            .getByText('Contract ' + contractID.slice(0, 5))
+        ).toBeVisible()
+      }
+
+      if (networkVersion === 'v2') {
```

Comment:
Instead of duplicating the test implementation for every network version, I'd write one implementation of the test and then use a switch on the network version where it's appropriate or needed. That way we don't needlessly duplicate a bunch of code but it also serves as documentation, highlighting the differences between the network versions.

Reply:
@telestrial similar to the discussions in the code for contracts/transactions, I think it would be most clear to just split out separate tests rather than weaving conditionals. It doesn't duplicate that much and will make it easier to leave the v1 test alone and continue to extend the v2 features and test in the future with no concern over v1.